Samples/Server/Unity-FractalStream/Assets/Scripts/StreamingUnityServerPlugin.cs

```diff
@@ -216,31 +223,38 @@ public StreamingUnityServerPlugin()
         }
 
         /// <summary>
-        /// The <see cref="CommandBuffer"/> needed to encode and transmit frame data
+        /// Backing field for <see cref="EncodingStereo"/>
         /// </summary>
-        /// <remarks>
-        /// This is consumed by <see cref="EncodeAndTransmitFrame"/> to encode the entire rendertexture
-        /// and can also be added to a camera command buffer to encode just that cameras view
-        /// </remarks>
-        /// <example>
-        /// var plugin = new StreamingUnityServerPlugin();
-        /// Camera.main.AddCommandBuffer(CameraEvent.AfterEverything, plugin.EncodeAndTransmitCommand);
-        /// </example>
-        public CommandBuffer EncodeAndTransmitCommand

```

Comment:
😢 this breaks the compos-ability of rendering here. the value here was the ability to choose what to encode as the "frame" - is there a reason we need to give this up to support timestamps, or can we rework this code to have both?

Reply:
Sorry Ben, I removed it since I couldn't find any usage of it after switching to rendering to texture. Could you please provide me an example of how you use it?

Reply:
there's an example in the xmldoc - the idea was to allow the caller to decide to encode and transmit data from the unity graphics pipeline where they desire, since they may not want our default behavior of "capture the entire view of a given camera after all other graphics events occur".

without this the caller isn't able to easily modify how and when our toolkit encodes and sends video data from the graphics pipeline (which was a feature, imo, of the initial design here).

Reply:
Hmm, now I need your help to understand more about its usage in our current master since it doesn't look right to me. My understanding is that **EncodeAndTransmitCommand** calls **GetRenderEventFunc** and then **OnEncode**. And in **OnEncode**, it doesn't do anything but initialization. Why do we need to call OnEncode every frame and do nothing? This confuses me so that's why I need more example to understand about this design. It would be great if you can modify my branch to demonstrate the usage.

Reply:
TLDR: we've been calling OnEncode each frame forever, i was under the impression that was responsible for reading from the unity render target and writing to our webrtc stream. i am no longer completely convinced this is what's happening, in which case i agree with you that this may no longer make sense.

---

i'm not sure i have a good enough understanding of what is actually happening here... it's been [that way](https://github.com/CatalystCode/3dtoolkit/blob/5b150972d966c9c2f4aaee7e2a800b350ed6b10d/Samples/Server/Unity-FractalStream/Assets/WebRTCServer.cs#L55) since [the beginning](https://github.com/CatalystCode/3dtoolkit/blob/5b150972d966c9c2f4aaee7e2a800b350ed6b10d/Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp#L139) and i'm not sure why. I do recall (granted, a while back, when i first investigated this stuff) trying to debug and seeing issues where `s_context` in `OnEncode` was `NULL` and it was causing problems. perhaps calling every frame was a workaround to "retry" until it isn't `NULL`?

my understanding (which looking at this deeper...may not be the case...looks like that would only occur is s_frameBuffer was nulled out at the end of each frame [which i don't see happening]) was that `OnEncode` was grabbing the render target each frame, and passing it down our pipeline.

Reply:
Me neither since I didn't spend much time on the Unity-FractacStream. Thanks a lot for the detailed explanation, Ben. I love your design though.